utils/Constants.sol

```diff
@@ -304,6 +299,17 @@ contract Constants {
         }
     }
 
+    function getCauldron(string calldata chain, string memory name, uint8 version) public view returns (CauldronInfo memory) {
+        CauldronInfo[] memory cauldronInfos = cauldronsPerChain[chain];
+
```

Comment:
here it's ok that it's not optimized since it's just for testing, just telling you for educational purposes but in live code what you do is a mapping so it's indexed by name for example.

so you can do things like cauldronsByName("WBTC") where cauldronsByName is `mapping(string => address)`, if it returns address(0) then it doesn't exists

could also be cauldronsByNameAndVersion `mapping(string => mapping(version => address))`, quite powerful and grows indefinitely inside the contract storage using keccak256 (iirc) hashing, pretty cool stuff.

Reply:
Gave this a shot!